docs/user_guide.md

```diff
@@ -0,0 +1,225 @@
+  in-network computing features to accelerate inter-node collectives.
+- `rccl`: TL leveraging [RCCL](https://github.com/ROCmSoftwarePlatform/rccl) for collectives
+  on ROCm device memory. 
+
+UCC is extensible so vendors can provide additional TLs. For example the UCC binaries shipped
+with [HPC-X](https://developer.nvidia.com/networking/hpc-x) add the `shm` TL with optimized
+CPU shared memory collectives.
+
+UCC exposes environment variables to tune CL and TL selection and behavior. The list of all
+environment variables with a description is available from `ucc_info`:
+
+```
+$ ucc_info -caf | head -15
+# UCX library configuration file
+# Uncomment to modify values
+
+#
+# UCC configuration
+#
+
+#
+# Comma separated list of CL components to be used
+#
+# syntax:    comma-separated list of: [basic|hier|all]
+#
+UCC_CLS=basic
+```
+
+In this guide we will focus on how TLs are selected based on a score. Every time UCC needs
+to select a TL the TL with the highest score is selected considering:
+
+- The collective type
+- The message size
+- The memory type
+- The team size (number of ranks participating in the collective)
+
+A user can set the `UCC_TL_<NAME>_TUNE` environment variables to override the default scores
+following this syntax:
+
+```
+UCC_TL_<NAME>_TUNE=token1#token2#...#tokenN,
+```
+
+Passing a `# ` separated list of tokens to the environment variable. Each token is a `:`
+separated list of qualifiers:
+
+```
+token=coll_type:msg_range:mem_type:team_size:score:alg
+```
+
+Where each qualifier is optional. The only requirement is that either `score` or `alg`
+is provided. The qualifiers are
+
+- `coll_type = coll_type_1,coll_type_2,...,coll_type_n` - a `,` separated list of
+  collective types.
+- `msg_range = m_start_1-m_end_1,m_start_2-m_end_2,..,m_start_n-m_end_n` - a `,`
+  separated list of msg ranges in byte, where each range is represented by `start`
+  and `end` values separated by `-`. Values can be integers using optional binary
+  prefixes. Supported prefixes are `K=1<<10`, `M=1<<20`, `G=1<<30` and, `T=1<<40`.
+  Parsing is case indepdent and a `b` can be optionally added. The special value
+  `inf` means MAX msg size. E.g. `128`, `256b`, `4K`, `1M` are valid sizes. 
+- `mem_type = m1,m2,..,mN` - a `,` separated list of memory types
+- `team_size = [t_start_1-t_end_1,t_start_2-t_end_2,...,t_start_N-t_end_N]` - a
+  `,` separated list of team size ranges enclosed with `[]`.
+- `score =` , a `int` value from `0` to `inf`
+- `alg = @<value|str>` - character `@` followed by either the `int` or string
+  representing the collective algorithm.
+
+Supported memory types are:
+- `cpu`: for CPU memory. 
+- `cuda`: for pinned CUDA Device memory (`cudaMalloc`).
+- `cuda_managed`: for CUDA Managed Memory (`cudaMallocManaged`).
+- `rocm`: for pinned ROCm Device memory.
+- `rocm_managed`: for ROCm Managed Memory.
+
```

Comment:
I would suggest to remove rocm_managed from this list, we have realistically only tested with rocm device memory.

Reply:
Agree if `rocm_managed` is effectively not supported (as not tested) I agree it should be dropped. Keeping this open to address it with the other points for which I need clarification.